Artisan/CraftingLogic/ExpertSolver/State.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Artisan.CraftingLogic.CraftData;
+
+namespace Artisan.CraftingLogic.ExpertSolver;
+
+public class CraftState
```

Comment:
Ok so this one could potentially be removed in favour of global state (like the rest of artisan does), but that would complicate running sims outside crafts (eg when tweaking options and doing A/B comparisons). I feel that this should be refactored somehow.